gcloud/monitoring/agent_translation_service_api.py

```diff
@@ -0,0 +1,143 @@
+        'https://www.googleapis.com/auth/monitoring.read',
+        'https://www.googleapis.com/auth/cloud-platform',
+    )
+
+
+
+    class Templates(object):
+        """PathTemplates for resources used by AgentTranslationServiceApi."""
+        PROJECT = PathTemplate.from_string(
+            'projects/{project}')
+
+    def __init__(
+            self,
+            service_path=_SERVICE_ADDRESS,
+            port=_DEFAULT_SERVICE_PORT,
+            channel=None,
+            ssl_creds=None,
+            scopes=_ALL_SCOPES,
+            is_idempotent_retrying=True,
+            max_attempts=3,
+            timeout=30):
+        self.defaults = api_callable.ApiCallableDefaults(
+            timeout=timeout,
+            max_attempts=max_attempts,
+            is_idempotent_retrying=is_idempotent_retrying)
+
+        self.stub = config.create_stub(
+            agent_service_pb2.beta_create_AgentTranslationService_stub,
+            service_path,
+            port,
+            ssl_creds=ssl_creds,
+            channel=channel,
+            scopes=scopes)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        del self.stub
+
+    # Service calls
+    def create_collectd_time_series(
+            self,
+            name='',
+            resource=None,
+            collectd_version='',
+            collectd_payloads=None,
+            **kwargs):
+        """
+        Creates a new time series with the given data points.  This method
+        is only for use in `collectd`-related code, including the Google
+        Monitoring Agent. See
+        [google.monitoring.v3.MetricService.CreateTimeSeries] instead.
+
+        :type name: string
+        :type resource: monitored_resource_pb2.MonitoredResource
+        :type collectd_version: string
+        :type collectd_payloads: list of agent_pb2.CollectdPayload
+        """
+        if resource is None:
+            resource = monitored_resource_pb2.MonitoredResource()
+        if collectd_payloads is None:
+            collectd_payloads = []
+        create_collectd_time_series_request = agent_service_pb2.CreateCollectdTimeSeriesRequest(
+            name=name,
+            resource=resource,
+            collectd_version=collectd_version,
+            collectd_payloads=collectd_payloads,
+            **kwargs)
+        return self.create_collectd_time_series_callable()(create_collectd_time_series_request)
+
+    def create_collectd_time_series_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None):
+        return api_callable.idempotent_callable(
+            self.stub.CreateCollectdTimeSeries,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    # ========
+    # Manually-added methods: add custom (non-generated) methods after this point.
+    # ========
```

Comment:
The red line means the file doesn't end in a newline. Your files should end in a newline. (I don't know why this is the convention, but it is.)